### ISSUE TITLE
support lxd clustering

### ISSUE THREAD

**Issue by bit** [lxc/pylxd#292]
i.e. no way to specify the target host in container.create.
in the rest api this is done by adding ?target=target_id

**Comment by ajkavanagh** [comment]
Yes, this is a 3.x feature added via the 'clustering' API extension.

The 'clustering' extension affects a number of areas: containers, storage-pools, and networks.

(Note for implementation: remember to check that the 'clustering' API extension exists as a check as to whether to accept this parameter.)

**Comment by ghoshanjega** [comment]
Any timeline on when clustering will be supported?

**Comment by felix-engelmann** [comment]
@ghoshanjega there is now an api endpoint for getting cluster and member info. What else would you need? 
@bit there is a target parameter for containers now.

**Comment by mkowalski** [comment]
Hi guys,

I have started using pylxd very recently and the first thing I noticed missing in the cluster supports is creation of networks. Where in the standalone LXD it's enough to do
```
lxc network create ...
```
in a clustered environment the process is a bit more complex
```
lxc network create ... --target node1
...
lxc network create ... --target nodeN
lxc network create ...
```
Any idea if we could get this supported?

**Comment by felix-engelmann** [comment]
The addition of a target argument for containers was very lightweight.
To support networks in cluster mode, I need the following information:
- @stgraber does the lxd api support an undocumented `?target=` query parameter for networks like the instance endpoint?
- @ajkavanagh what is the architecture decision on the abstraction of this library: Make one complex call which manages the network creation on all nodes or leave it up to the user to perform the necessary calls and add a simple target argument?

**Comment by stgraber** [comment]
@felix-engelmann `?target` for networks and storage-pools is supported and lets you create/set the host-specific config ahead of adding a network or storage-pool globally.

That's why you usually need to create a network like:
```
lxc network create blahbr0 --target a
lxc network create blahbr0 --target b
lxc network create blahbr0 --target c
lxc network create blahbr0
```

The targeted ones can set keys that are only relevant to a particular member, in the network case, that's pretty much just `bridge.external_interfaces`.
